app/helpers/user_profiles_helper.rb

```diff
@@ -0,0 +1,29 @@
+module UserProfilesHelper
+  def resource
+    @resource ||= current_user
+  end
+
+  def resource_name
+    current_user.user_profile.name || :user
+  end
+
+  def devise_mapping
+    @devise_mapping ||= Devise.mappings[:user]
+  end
+
+  def show_name?
```

Comment:
usually method names with ? at the end of name are used when method supposed to return true/false, not some real data.